eng/common/TestResources/New-TestResources.ps1

```diff
@@ -52,12 +33,30 @@ param (
     [string] $Location = '',
 
     [Parameter()]
-    [ValidateSet('AzureCloud', 'AzureUSGovernment', 'AzureChinaCloud')]
     [string] $Environment = 'AzureCloud',
 
     [Parameter()]
     [hashtable] $AdditionalParameters,
 
+    [Parameter(ParameterSetName = 'SubscriptionConfiguration', Mandatory = $true)]
+    [string] $SubscriptionConfiguration = '',
+
+    [Parameter(ParameterSetName = 'SubscriptionConfiguration', Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [string] $KeyVaultName,
+
+    [Parameter(ParameterSetName = 'SubscriptionConfiguration', Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [string] $KeyVaultTenantId,
```

Comment:
I debated re-using the `ProvisoinerAppId` and friends here but it's wrong to read that variable from the invocation and then rewrite it with a new value.

(there is some hypocrisy here: we do figure out the value of `$Location` and overwrite it)